pages/_app.js

```diff
@@ -2,10 +2,48 @@ import GlobalStyle from "../styles";
 import { useState } from "react";
 import Layout from "@/components/Layout/Layout";
 import { medications } from "@/lib/db";
+import { uid } from "uid";
+import { useImmerLocalStorageState } from "../lib/hook/useImmerLocalStorageState";
+import ConfirmationModal from "@/components/ConfirmationModal/ConfirmationModal";
 
 export default function App({ Component, pageProps }) {
-  const [medicationsList, setMedicationsList] = useState(medications);
+  const [medicationsList, setMedicationsList] = useImmerLocalStorageState(
+    "medicationsList",
+    {
+      defaultValue: medications,
+    }
+  );
+  const [medicationPlan, setMedicationPlan] = useImmerLocalStorageState(
+    "medicationPlan",
+    {
+      defaultValue: [],
+    }
+  );
+  const [forSomeoneElse, setForSomeoneElse] = useState(false);
+  const [showConfirmation, setShowConfirmation] = useState(false);
+  const handleForChange = (event) => {
+    setForSomeoneElse(event.target.value === "Someone else");
+  };
 
+  const handleAddMedicationPlanSubmit = (event) => {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    let frequency = {};
+    for (let [key, value] of formData.entries()) {
+      if (key.startsWith("frequency")) {
+        const field = key.split("frequency")[1];
+        frequency[field] = value;
+      }
+    }
+    const plan = {
+      ...Object.fromEntries(formData.entries()),
```

Comment:
I think we can just write 

``` js
const plan = {
  ...Object.fromEntries(formData),
  frequency,
  id: uid(),
};
```

This is because the `formData` object is already an iterable and does not need to call the `.entries()` method.

Reply:
thank you for your helpful comment , i have already apply all suggested chenges.